lib/ansible/modules/system/user.py

```diff
@@ -1501,6 +1509,7 @@ class DarwinUser(User):
         ('shell', 'UserShell'),
         ('uid', 'UniqueID'),
         ('group', 'PrimaryGroupID'),
+        ('hidden', 'IsHidden'),
```

Comment:
As you noted, we'll need to handle hidden inside of the modify_user function with new code instead of relying on looping through self.fields for it.